### PR TITLE
created new file rcalc-keypress.js, and new functionality, program ta…

### PR DESCRIPTION
**Pull request by romelu**
…kes keyboard inputs like 0-9 numerical keys and mi(-), mu(*), di(/), pl(+), e(evaluates) and c(clear)

### CODE REVIEW

File: code/js/rcalc-keypress.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    displayNumber("*");
+    isSecondPressed = false;
+  }
+  if (nextToLastCharacter == "d" && key == "i" && isSecondPressed == true) {
+    displayNumber("/");
+    setWarning(key);
+    removeExtraZero();
+    isSecondPressed = false;
+  } else {
+    isSecondPressed = false;
```
Comment (palomino7) [inline code/js/rcalc-keypress.js]: Command: isSecondPressed = false; you call in each IF statement and also in ELSE statement. This means that our will do always ... so remove it from all IF and remove ELSE statement and just put this command in the last row of this method.

File: code/js/rcalc-keypress.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    case "c":
+      clearDisplay();
+      break;
+  }
+});
+
+function removeExtraZero() {
+  var diplayValue = document.getElementById("id-display-main").value;
+  diplayValue = diplayValue.slice(0, -1);
+  document.getElementById("id-display-main").value = diplayValue;
```
Comment (palomino7) [inline code/js/rcalc-keypress.js]: id-display-main .. should be in constant file. Please do it like this that you will call this id this way> html-id-constant.DISPLAY_MAIN  ... fix it ids in all places for all ids. 
It is possible do it file name dependent as I wrote here?

File: code/js/rcalc-keypress.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+document.addEventListener("keyup", function(event) {
+  if (event.defaultPrevented) {
+    return;
+  }
+  var key = event.key;
+  collectionOfKeys.push(key);
+  nextToLastCharacter = collectionOfKeys[collectionOfKeys.length - 2];
+  switch (key) {
+    case "0":
```
Comment (palomino7) [inline code/js/rcalc-keypress.js]: For many keys you are do same thing. You don't have to write same code, just write list of keys and then same action.
case "1":
case "2":
your action;
break;

You save many rows.

File: code/js/rcalc-keypress.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+var collectionOfKeys = [];
+var nextToLastCharacter;
+
+document.addEventListener("keyup", function(event) {
+  if (event.defaultPrevented) {
+    return;
+  }
+  var key = event.key;
+  collectionOfKeys.push(key);
+  nextToLastCharacter = collectionOfKeys[collectionOfKeys.length - 2];
```
Comment (palomino7) [inline code/js/rcalc-keypress.js]: Why minus -2? Specify or put comment. Magic number man.

File: code/js/rcalc-keypress.js (hunk 3 above)
Comment (palomino7) [inline code/js/rcalc-keypress.js]: Please create constant file for all your checked keys. 
rcalc-keys.OPERATOR_PLUS
rcalc-keys.NUMBER_1
rcalc-keys.KEY_M
...

File: code/js/rcalc-keypress.js [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+      evaluateDisplay();
+      break;
+    case "c":
+      clearDisplay();
+      break;
+  }
+});
+
+function removeExtraZero() {
+  var diplayValue = document.getElementById("id-display-main").value;
```
Comment (palomino7) [inline code/js/rcalc-keypress.js]: typo in diplayValue

File: code/js/rcalc-keypress.js [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+  diplayValue = diplayValue.slice(0, -1);
+  document.getElementById("id-display-main").value = diplayValue;
+}
+
```
Comment (palomino7) [inline code/js/rcalc-keypress.js]: WTF ... setBoolean is nothing to say name of method!!!!!! This is something like doCode, doCommand, doSomething.

File: code/js/rcalc-keypress.js [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+function setBoolean() {
+  isSecondPressed = true;
+}
+
+function testCharacter(key) {
+  if (nextToLastCharacter == "p" && key == "l" && isSecondPressed == true) {
```
Comment (palomino7) [inline code/js/rcalc-keypress.js]: You have to write isSecondPressed == true ... you can't write just: isSecondPressed

File: code/js/rcalc-keypress.js (hunk 1 above)
Comment (romelu) [reply code/js/rcalc-keypress.js]: isSecondPressed = false is on the end of the method only

File: code/js/rcalc-keypress.js (hunk 3 above)
Comment (romelu) [reply code/js/rcalc-keypress.js]: keys are listed as adviced

File: code/js/rcalc-keypress.js (hunk 4 above)
Comment (romelu) [reply code/js/rcalc-keypress.js]: explanation of why -2 in the comment

File: code/js/rcalc-keypress.js (hunk 5 above)
Comment (romelu) [reply code/js/rcalc-keypress.js]: typo fixed

File: code/js/rcalc-keypress.js (hunk 7 above)
Comment (romelu) [reply code/js/rcalc-keypress.js]: This one is correct, somehow you overlooked that there is  isSecondPressed == true

File: code/js/rcalc-keypress.js (hunk 1 above)
Comment (romelu) [reply code/js/rcalc-keypress.js]: isSecondPressed  boolean is not needed at all to make the program work... it has been completely removed

File: code/js/rcalc-keypress.js (hunk 6 above)
Comment (romelu) [reply code/js/rcalc-keypress.js]: setBoolean() has been completely removed as it is not needed... not necessary extra code

File: code/js/rcalc-keypress.js (hunk 7 above)
Comment (romelu) [reply code/js/rcalc-keypress.js]: pointless now....  isSecondPressed == true has been removed... not needed at all, unnecessary code

File: code/js/rcalc-keypress.js (hunk 2 above)
Comment (romelu) [reply code/js/rcalc-keypress.js]: HTML_ID_CONST.ID_DISPLAY_MAIN this is how to access "id-display-main"... fixed

File: code/js/rcalc-keypress.js (hunk 3 above)
Comment (romelu) [reply code/js/rcalc-keypress.js]: created file string-key-keypress-constant.js ... all keys used in this functionality are constants now

File: code/js/rcalc-keypress.js (hunk 4 above)
Comment (palomino7) [reply code/js/rcalc-keypress.js]: number 2 should be replaced some constant and this constant should have comment

**Pull request merged**